Assets/Scripts/Game/Entities/PlayerEntity.cs

```diff
@@ -460,56 +461,43 @@ public override void Update(DaggerfallEntityBehaviour sender)
-            //        if (DFRandom.random_range_inclusive(10, 100) < 30)
-            //            Questing.QuestMachine.Instance.InstantiateQuest("$CUREVAM");
-            //    }
-            //    else if (hasStartedInitialVampireQuest)
-            //    {
-            //        // Do a quest starting with "P0B" for player's level
-            //        // TODO: Merge with rest of quest-loading code
-            //        string[] vampireQuests = { "P0B00L01", "P0B00L03", "P0B00L04", "P0B00L06", "P0B01L02", "P0B10L07", "P0B10L08", "P0B10L10", "P0B20L09" };
-            //        List<string> vQList = new List<string>(vampireQuests);
-            //        if (DFRandom.random_range_inclusive(1, 100) < 50)
-            //        {
-            //            bool found = false;
-            //            string choice = string.Empty;
-            //            while (!found && vQList.Count > 0)
-            //            {
-            //                int index = UnityEngine.Random.Range(0, vQList.Count);
-            //                choice = vQList[index];
-            //                // 4th letter of name is the minimum level the player must be to get the quest
-            //                if (choice[3] - 48 <= level)
-            //                {
-            //                    found = true;
-            //                }
-            //                else
-            //                    vQList.RemoveAt(index);
-            //            }
-
-            //            if (found)
-            //                Questing.QuestMachine.Instance.InstantiateQuest(choice);
-            //        }
-            //    }
-            //    else if (DFRandom.random_range_inclusive(1, 100) < 50)
-            //    {
-            //        Questing.QuestMachine.Instance.InstantiateQuest("P0A01L00");
-            //        hasStartedInitialVampireQuest = true;
-            //    }
-            //}
-            ///*else
-            //{
-            //    if (playerIsWereCreature && isCureQuest && DFRandom.random_range_inclusive(1, 100) < 30)
-            //    {
-            //        Questing.QuestMachine.Instance.InstantiateQuest("$CUREWER");
-            //    }
-            //}*/
+            if (Race == Races.Vampire)
+            {
+                if (isCureQuest)
+                {
+                    if (DFRandom.random_range_inclusive(10, 100) < 30)
+                        QuestMachine.Instance.InstantiateQuest("$CUREVAM");
+                }
+                else if (hasStartedInitialVampireQuest)
+                {
+                    // Get an appropriate quest for player's level?
+                    if (DFRandom.random_range_inclusive(1, 100) < 50)
+                    {
+                        // Get the regional vampire clan faction id for affecting reputation on success/failure, and current rep
+                        int factionId = 23; // TODO: get appropriate value - just hardcoding The Vraseth for now!
```

Comment:
I should point out this line which needs the vampire clan faction id for doling out rep gains etc upon quest completion

Reply:
The clan can be found from VampirismEffect incumbent after player turns, or from VampirismInfection prior to turning. This can be pulled now if you want to set it up. Vampire faction IDs are 150-158 and have a matching enum in DaggerfallUnityEnums.VampireClans.

Reply:
I will leave this for you to add when you're ready to move the code.